README.md
# Murotal

High Quality Murotal Qur'an from Archive.org

![all contributors](https://img.shields.io/github/contributors/zeroc0d3/murotal)
![tags](https://img.shields.io/github/v/tag/zeroc0d3/murotal?sort=semver)
![issues](https://img.shields.io/github/issues/zeroc0d3/murotal)
![pull requests](https://img.shields.io/github/issues-pr/zeroc0d3/murotal)
![forks](https://img.shields.io/github/forks/zeroc0d3/murotal)
![stars](https://img.shields.io/github/stars/zeroc0d3/murotal)
[![License: CC BY-NC 4.0](https://img.shields.io/badge/License-CC%20BY--NC%204.0-blue.svg)](https://creativecommons.org/licenses/by-nc/4.0/)

## Per-Juz (30 Juz) - Categories

```
1000. Syeikh Misyari Rasyid
1200. Syeikh Saad Al-Ghamidi
1400. Syeikh Abdurrahman As-Sudais
1600. Syeikh Hany Ar-Rifa'i
1800. Syeikh Yasser Al-Dosari
2000. Syeikh Saud Al-Shuraim
2200. Syeikh Nasser Al-Qatami
2400. Syeikh Abdul Basith
```

## How-to-Use

- [Index Documentation](docs/INDEX.md)

## Copyright
- Audio: **To the respect owner**
- Script Author: **Dwi Fahni Denni (@zeroc0d3)**
- License: **Creative Commons (Non Commercial)**
  <br><a rel="license" href="http://creativecommons.org/licenses/by/4.0/"><img alt="Lisensi Creative Commons" style="border-width:0;max-width:100px" src="https://upload.wikimedia.org/wikipedia/commons/thumb/9/99/Cc-by-nc_icon.svg/320px-Cc-by-nc_icon.svg.png" /></a><br />This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.

CHANGELOG.md
# Change Log History (Murotal)

High Quality Murotal Qur'an from Archive.org

## Version 0.1

### Per-Juz (30 Juz)

- 01. Syeikh Misyari Rasyid
- 02. Syeikh Saad Al-Ghamidi
- 03. Syeikh Abdurrahman As-Sudais
- 04. Syeikh Hany Ar-Rifa'i
- 05. Syeikh Yasser Al-Dosari
- 06. Syeikh Saud Al-Shuraim
- 07. Syeikh Nasser Al-Qatami
- 08. Syeikh Abdul Basith

## Version 0.2

### Per-Juz (30 Juz) - Categories

```
1000. Syeikh Misyari Rasyid
1200. Syeikh Saad Al-Ghamidi
1400. Syeikh Abdurrahman As-Sudais
1600. Syeikh Hany Ar-Rifa'i
1800. Syeikh Yasser Al-Dosari
2000. Syeikh Saud Al-Shuraim
2200. Syeikh Nasser Al-Qatami
2400. Syeikh Abdul Basith
```

## Version 0.3

- Refactoring script downloader
- Added skip sequence for existing file
- Added documentation how to use in `docs/index.md`
- Added test script

## Version 0.4

- Added autodownload script for selected ID Murotal
  ```
  ./autodownload.sh [MID] [PATH_FOLDER]
  ---
  eg:
  ./autodownload.sh 10 /home/zeroc0d3/Documents/murotal     # Download all murotal Syeikh Misyari Rasyid and save to spesific folder
  ./autodownload.sh 16 /home/zeroc0d3/Documents/murotal     # Download all murotal Syeikh Hani Ar-Rifai and save to spesific folder
  ```
- Fixing wrong path download from MID: 2400
  ```
  PATH_MUROTAL="[PATH_DOWNLOAD]/2400_Abdul_Basit_AbdulSamad"
  ```

autodownload.sh
#!/bin/sh

set -e

TITLE="MUROTAL DOWNLOADER SCRIPT"    # script name
VER="2.2"                            # script version

ID_CODE=$1
TARGET_PATH=$2

COL_RED="\033[22;31m"
COL_GREEN="\033[22;32m"
COL_BLUE="\033[22;34m"
COL_END="\033[0m"

get_time() {
  DATE=`date '+%Y-%m-%d %H:%M:%S'`
}

print_line0() {
  echo "$COL_GREEN=====================================================================================$COL_END"
}

print_line1() {
  echo "$COL_GREEN-------------------------------------------------------------------------------------$COL_END"
}

print_line2() {
  echo "-------------------------------------------------------------------------------------"
}

logo() {
  clear
  print_line0
  echo "$COL_RED'########:'########:'########:::'#######:::'######::::'#####:::'########:::'#######:: $COL_END"
  echo "$COL_RED..... ##:: ##.....:: ##.... ##:'##.... ##:'##... ##::'##.. ##:: ##.... ##:'##.... ##: $COL_END"
  echo "$COL_RED:::: ##::: ##::::::: ##:::: ##: ##:::: ##: ##:::..::'##:::: ##: ##:::: ##:..::::: ##: $COL_END"
  echo "$COL_RED::: ##:::: ######::: ########:: ##:::: ##: ##::::::: ##:::: ##: ##:::: ##::'#######:: $COL_END"
  echo "$COL_RED:: ##::::: ##...:::: ##.. ##::: ##:::: ##: ##::::::: ##:::: ##: ##:::: ##::...... ##: $COL_END"
  echo "$COL_RED: ##:::::: ##::::::: ##::. ##:: ##:::: ##: ##::: ##:. ##:: ##:: ##:::: ##:'##:::: ##: $COL_END"
  echo "$COL_RED ########: ########: ##:::. ##:. #######::. ######:::. #####::: ########::. #######:: $COL_END"
  echo "$COL_RED........::........::..:::::..:::.......::::......:::::.....::::........::::.......::: $COL_END"
  print_line1
  echo "$COL_GREEN# $TITLE :: ver-$VER $COL_END"
}

header() {
  logo
  print_line0
  get_time
  echo "$COL_RED# BEGIN PROCESS..... (Please Wait)  $COL_END"
  echo "$COL_RED# Start at: $DATE  $COL_END"
}

footer() {
  print_line0
  get_time
  echo "$COL_RED# Finish at: $DATE  $COL_END"
  echo "$COL_RED# END PROCESS.....  $COL_END\n"
}

download_murotal_perjuz() {
  mkdir -p $PATH_MUROTAL
  cd $PATH_MUROTAL

  counter=1

  while [ $counter -le 30 ]
  do
      if [ $counter -lt 10 ]
      then
        download_below10 $counter
      else
        download_upper10 $counter
      fi
      ((counter++))
  done
}

selected_download() {
  case $ID_CODE in
    10 | 1000)
      export PATH_MUROTAL="$TARGET_PATH/1000_Misyari_Rasyid"
      export DOWNLOAD_PATH="https://ia801204.us.archive.org/17/items/Misyari_Rasyid_per_Juz"
      ;;

    12 | 1200)
      export PATH_MUROTAL="$TARGET_PATH/1200_Saad_AlGhamidi"
      export DOWNLOAD_PATH="https://ia903200.us.archive.org/20/items/SaadAl-GhamidiPerJuz"
      ;;

    14 | 1400)
      export PATH_MUROTAL="$TARGET_PATH/1400_Abdurrahman_AsSudais"
      export DOWNLOAD_PATH="https://ia801904.us.archive.org/14/items/AbdurrahmanSudaisPerJuz"
      ;;

    16 | 1600)
      export PATH_MUROTAL="$TARGET_PATH/1600_Hany_ArRifai"
      export DOWNLOAD_PATH="https://ia801207.us.archive.org/10/items/HanyAr-RifaiPerJuz"
      ;;

    18 | 1800)
      export PATH_MUROTAL="$TARGET_PATH/1800_Yasser_AlDosari"
      export DOWNLOAD_PATH="https://ia803203.us.archive.org/26/items/YasserAlDosariPerJuz"
      ;;

    20 | 2000)
      export PATH_MUROTAL="$TARGET_PATH/2000_Saud_AlShuraim"
      export DOWNLOAD_PATH="https://ia903208.us.archive.org/20/items/SaudAlShuraimPerJuz"
      ;;

    22 | 2200)
      export PATH_MUROTAL="$TARGET_PATH/2200_Nasser_AlQatami"
      export DOWNLOAD_PATH="https://ia800400.us.archive.org/23/items/Nasseralqatamieveryjuz"
      ;;

    24 | 2400)
      export PATH_MUROTAL="$TARGET_PATH/2400_Abdul_Basit_AbdulSamad"
      export DOWNLOAD_PATH="https://ia903200.us.archive.org/23/items/AbdulBasitAbdulsamadPerJuz"
      ;;

    *)
      echo ""
      echo ">> Unknown MID (Murotal ID)"
      echo ">> Set to default MID: 10"
      export PATH_MUROTAL="$TARGET_PATH/1000_Misyari_Rasyid"
      export DOWNLOAD_PATH="https://ia801204.us.archive.org/17/items/Misyari_Rasyid_per_Juz"
      echo ""
      ;;
  esac
  download_murotal_perjuz
}

download_below10() {
  JUZ="0$1"
  DOWNLOAD_LINK="$DOWNLOAD_PATH/$JUZ.mp3"
  FNAME="$PATH_MUROTAL/$ID_CODE$JUZ.mp3"
  msg_download $JUZ $DOWNLOAD_LINK
  skip_exists $FNAME $DOWNLOAD_LINK
  echo "- DONE - "
  echo ""
}

download_upper10() {
  JUZ="$1"
  DOWNLOAD_LINK="$DOWNLOAD_PATH/$JUZ.mp3"
  FNAME="$PATH_MUROTAL/$ID_CODE$JUZ.mp3"
  msg_download $JUZ $DOWNLOAD_LINK
  skip_exists $FNAME $DOWNLOAD_LINK
  echo "- DONE - "
  echo ""
}

msg_download() {
  print_line2
  get_time
  echo "$COL_BLUE[ $DATE ] ##### Download Juz: $1 "
  echo "$COL_GREEN[ $DATE ]       wget $2 $COL_END"
  get_time
  print_line2
}

skip_exists() {
  if [[ -f "$1" ]]
  then
    echo ">> Skip for existing file $1 ..."
  else
    # echo "wget -O $1 $2"
    wget -O $1 $2
  fi
}

main() {
  header
  selected_download $1 $2
  footer
}

### START HERE ###
main $1 $2
